pyomo/contrib/multistart/multi.py

```diff
@@ -170,7 +170,9 @@ def solve(self, model, **kwds):
                 result = solver.solve(m, **config.solver_args)
                 if (result.solver.status is SolverStatus.ok and
                         result.solver.termination_condition is tc.optimal):
-                    obj_val = value(m.obj.expr)
+                    model_objectives = m.component_data_objects(Objective, active=True)
```

Comment:
I believe this is also explicitly checked around line 122.

Reply:
I see you are verifying that there is a single objective above.  That technically makes the `None` in `next(model_objectives, None)` redundant and slightly confusing, as it implies that you are expecting situations where `next()` might not have an object to return (which leads others to wonder what that situation might be).  You are not checking for None, so if that were ever to be the case, the user would see an exception `'NoneType' has no attribute 'expr'`.  The alternative of omitting the `None` would have the user / developer see the (possibly) more helpful `StopIteration` exception.

It is something analogous to using `{dict}.get('foo', None)` instead of `{dict}['foo']`.

Ass said and done, I suppose this comment is more stylistic than substantive.